python/rmm/rmm.py

```diff
@@ -267,6 +267,31 @@ def get_ipc_handle(ary, stream=0):
     )
 
 
+def make_cupy_use_rmm():
+    """
+    Make CuPy use RMM.
+    Requires the cupy module.
+    """
+    import cupy
+    from rmm._lib.device_buffer import DeviceBuffer
+
+    class RMMMemory(cupy.cuda.memory.BaseMemory):
+        def __init__(self, size):
+            self.size = size
+            self.device_id = cupy.cuda.device.get_device_id()
+            if size > 0:
+                self.rmm_array = DeviceBuffer(size=size)
+                self.ptr = self.rmm_array.ptr
+            else:
+                self.rmm_array = None
+                self.ptr = 0
```

Comment:
Would be good to pull this class definition out of the function. Though we could define it conditionally (only if we can `import cupy`).